---
title: "Gene Expression Project"
author: "Fenny and Daniëlle"
date: '2023-04-26'
output:
  html_document: default
  pdf_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(cache = TRUE)
```

# Hippocampal subfield transcriptome analysis in schizophrenia psychosis
```{r}
library(tidyr)
library(dplyr)
library(affy)
library(scales)
library(ggplot2)
library('DESeq2')
library('PoiClaClu')
library(knitr)
```

The count data is loaded into R.

Table 1: Raw counts data
```{r}
count_data <- "Sample_Subfields_Counts.txt"
counts <- read.table(file = count_data, header = TRUE)
head(counts)



dim(counts)
#str(counts)

control_CA1 <- c(1:13)
case_CA1 <- c(14:26)
control_CA3 <- c(27:39)
case_CA3 <- c(40:52)
control_DG <- c(53:65)
case_DG <- c(66:78)


counts_CA1 <- counts[, 1:26]
#counts_CA1
counts_CA3 <- counts[, 27:52]
#counts_CA3
counts_DG <- counts[, 53:78]
#counts_DG
```

The data of the counts was split based on the hippocampal subfield and tidied using tidyr (table 2-4).

Table 2: Tidy CA1 data
```{r}
counts_CA1_tidy <- pivot_longer(data = counts_CA1,
             cols=1:26,
             #names_pattern="(CA1_CTL|CA1_SZ).",
             names_to ="sample",
             values_to = "value")
head(counts_CA1_tidy)
```

Table 3: Tidy CA3 data
```{r}
counts_CA3_tidy <- pivot_longer(data = counts_CA3,
             cols=1:26,
             #names_pattern="(CA1_CTL|CA1_SZ).",
             names_to ="sample",
             values_to = "value")
head(counts_CA3_tidy)
```

Table 4: Tidy DG data
```{r}
counts_DG_tidy <- pivot_longer(data = counts_DG,
             cols=1:26,
             #names_pattern="(CA1_CTL|CA1_SZ).",
             names_to ="sample",
             values_to = "value")
head(counts_DG_tidy)
```

The data is separated in control and case groups for each subfield and visualized by creating a boxplot for each group (figure 1).

```{r}
control_ca1 <- c(1:13)
case_ca1 <- c(14:26)
control_ca3 <- c(27:39)
case_ca3 <- c(40:52)
control_dg <- c(53:65)
case_dg <- c(66:78)
```
```{r, fig.cap = "Figure 1: Raw counts boxplots", out.width = "100%", fig.height=8}
par(mfcol = c(3,2))
par(cex = 0.5)
if (identical(knitr::pandoc_to(), 'latex')) knit_hooks$set(plot = hook_plot_tex)
boxplot(log2(counts[control_ca1] + 1), outline = T, col = "green", las = 2, main = "A. CA1 Control", adj = 0)
boxplot(log2(counts[control_ca3] + 1), outline = T, col = "grey", las = 2, main = "C. CA3 Control", adj = 0)
boxplot(log2(counts[control_dg] + 1), outline = T, col = "red", las = 2, main = "E. DG Control", adj = 0)
boxplot(log2(counts[case_ca1] + 1), outline = T, col = "blue", las = 2, main = "B. CA1 Case", adj = 0)
boxplot(log2(counts[case_ca3] + 1), outline = T, col = "cyan", las = 2, main = "D. CA3 Case", adj = 0)
boxplot(log2(counts[case_dg] + 1), outline = T, col = "yellow", las = 2, main = "F. DG Case", adj = 0)
```

From the top down, two plots of subfields CA1, CA3 and DG are shown with the control groups on the left and the case groups on the right. 

A density plot is made for all the samples to visualize the distribution of the log2 of the count data.
```{r, fig.cap="Figure 2: Density plot of count distribution for all samples"}
myColors <- hue_pal()(4)

plotDensity(log2(counts + 0.1), col=rep(myColors, each=3),
            lty=c(1:ncol(counts)), xlab='Log2(count)',
            main='Expression Distribution')

legend('topright', names(counts), lty=c(1:ncol(counts)),
       col=rep(myColors, each=3))
abline(v=-1.5, lwd=1, col='red', lty=2)
```

Barplots are created for each control and case group in which the total raw count is shown of each sample.

```{r, fig.cap="Figure 3: Barplot showing the total raw reads count per million for each sample", fig.subcap=c("CA1 Control", "CA1 Case", "CA3 Control", "CA3 Case", "DG Control", "DG Case"), out.width = "100%", fig.height=8}
par(mfcol = c(3, 2))
barplot(colSums(counts[control_ca1]) / 1e6, col = "#D6F6DD", las = 2, cex.names = 0.8, main = "A. CA1 Control", adj = 0)
barplot(colSums(counts[control_ca3]) / 1e6, col = "#CCCCCC", las = 2, cex.names = 0.8, main = "C. CA3 Control", adj = 0)
barplot(colSums(counts[control_dg]) / 1e6, col = "#F4989C", las = 2, cex.names = 0.8, main = "E. DG Control", adj = 0)
barplot(colSums(counts[case_ca1]) / 1e6, col = "#DAC4F7", las = 2, cex.names = 0.8, main = "B. CA1 Case", adj = 0)
barplot(colSums(counts[case_ca3]) / 1e6, col = "#ACECF7", las = 2, cex.names = 0.8, main = "D. CA3 Case", adj = 0)
barplot(colSums(counts[case_dg]) / 1e6, col = "#EBD2B4", las = 2, cex.names = 0.8, main = "F. DG Case", adj = 0)
```

The plots show the CA1, CA3 and DG subfields from the top down, and the control groups are displayed on the left while the case groups are displayed on the right. The numbers on the y-axis indicate the total number of reads per million for each sample. The sample names can be found on the x-axis. For the majority of the reads, approximately 40 to 60 million reads were found. However, at least 10 million reads were found for every sample except for sample DG_SZ_0841 in figure 3F, which had around 5 million reads.

```{r}
#annotation <- data.frame(subfield = factor(rep(1:3, each = 26), 
 #                                         labels = c("CA1", "CA3", "DG")),
  #                       type = factor(rep(rep(1:2, each = 13), 3),
   #                                      labels = c("control", "schizophrenia")))
                         
# Set the rownames of the annotation dataframe to the sample names (required)
#rownames(annotation) <- names(counts)
```

The data is annotated in a dataframe with names containing control and case samples, and whether the case samples were derived from subjects on or off medication. This also enabled the identification of the sample types.
```{r}
load("annotation.RData")
```

Several DESeq2 functions are utilized to transform the raw read count data and calculating sample distances. A DESeqDataSet is made using the count data and the annotation which consists of the sample names.
```{r}
(ddsMat <- DESeqDataSetFromMatrix(countData = counts,
                                  colData = data.frame(samples = names(counts)),
                                  design = ~ 1))

# Perform normalization
#rld.dds <- vst(ddsMat)
# 'Extract' normalized values
#rld <- assay(rld.dds)
#sampledists <- dist( t( rld ))
#head(sampledists)
```

To visualize the calculated distances, Multi-Dimensional Scaling (MDS) is done, which uses Poisson distances. These distances are calculated using PoiClaClu functions. To do this, the count data is extracted from the DESeq dataset using the assay function. The t-function is used to transpose the matrix of the values, and the Poisson distance is calculated using the PoissonDistance function, which also normalizes the data.
```{r}

# Note: uses the raw-count data, PoissonDistance performs normalization
# set by the 'type' parameter (uses DESeq)
dds <- assay(ddsMat)
poisd <- PoissonDistance( t(dds), type = "deseq")
# Extract the matrix with distances
samplePoisDistMatrix <- as.matrix(poisd$dd)
# Calculate the MDS and get the X- and Y-coordinates
mdsPoisData <- data.frame( cmdscale(samplePoisDistMatrix) )

# And set some better readable names for the columns
names(mdsPoisData) <- c('x_coord', 'y_coord')
```

The annotated data is separated into groups depending on whether the sample was obtained from a control or case subject, whether they were on or off medication, and the hippocampal subfield of interest (CA1, CA3 or DG). These groups were labeled accordingly and used to plot a graph showing the MDS data.
```{r, fig.cap="Figure 4: Plot showing Multi-Dimensional Scaling of Poisson Distance for all samples"}
groups <- factor(rep(1:6, each=13), 
                 labels = c("CA1_CTL", "CA1_SZ", "CA3_CTL", "CA3_SZ", "DG_CTL", "DG_SZ"))
coldata <- names(counts)

color_palette <- c("#D6F6DD", "#DAC4F7", "#CCCCCC", "#ACECF7", "#F4989C", "#EBD2B4")

# Create the plot using ggplot

ggplot(mdsPoisData, aes(x_coord, y_coord, color = groups, label = ".")) + 
  geom_text(size = 10) +
  scale_colour_manual(values=c("green", "blue", "grey", "cyan", "red", "yellow"))+
  ggtitle('Multi Dimensional Scaling') +
  labs(x = "Poisson Distance", y = "Poisson Distance") +
  theme_bw()+
  labs(caption = "Figure: MDS plot")
```
All samples were plotted around the same area ex

A dataframe without sample CA3_CTL_0885 is made and the previous steps for the visualization in the MDS plot were repeated.
```{r}
count_nooutlier <- counts[ ,-38]
col1 <-rep(c("CA1", "CA3", "DG"), times = c(26, 25, 26))
col2 <-rep(c("control", "sz", "control", "sz", "control", "sz"), times = c(13, 13, 12, 13, 13, 13))
 
annotation_rm_ctl12 <- data.frame(col1, col2)
                        
rownames(annotation_rm_ctl12) <- names(count_nooutlier)
```

```{r}
(ddsMat_rm_CA3_CTL_12 <- DESeqDataSetFromMatrix(countData = count_nooutlier,
                                  colData = data.frame(samples = names(count_nooutlier)),
                                  design = ~ 1))

# Perform normalization
rld.dds <- vst(ddsMat)
# 'Extract' normalized values
rld <- assay(rld.dds)
sampledists <- dist( t( rld ))
```

```{r}
#annotation_rm_CA3
```

```{r}

# Note: uses the raw-count data, PoissonDistance performs normalization
# set by the 'type' parameter (uses DESeq)
dds <- assay(ddsMat_rm_CA3_CTL_12)
poisd <- PoissonDistance( t(dds), type = "deseq")
# Extract the matrix with distances
samplePoisDistMatrix <- as.matrix(poisd$dd)
# Calculate the MDS and get the X- and Y-coordinates
mdsPoisData_rm_CA3_CTL_12 <- data.frame( cmdscale(samplePoisDistMatrix) )

# And set some better readable names for the columns
names(mdsPoisData_rm_CA3_CTL_12 ) <- c('x_coord', 'y_coord')
```

```{r}
groups <- factor(rep(c("CA1_CTL", "CA1_SZ", "CA3_CTL", "CA3_SZ", "DG_CTL", "DG_SZ"), times = c(13, 13, 12, 13, 13, 13)))
coldata <- names(count_nooutlier)


# Create the plot using ggplot
ggplot(mdsPoisData_rm_CA3_CTL_12, aes(x_coord, y_coord, color = groups, label = ".")) + 
  geom_text(size = 20) +
  scale_colour_manual(values=c("green", "blue", "grey", "cyan", "red", "yellow"))+
  ggtitle('Multi Dimensional Scaling') +
  labs(x = "Poisson Distance", y = "Poisson Distance") +
  theme_bw()+
  labs(caption = "Figure: MDS plot with outlier removed")
```
Figure 15: Plot showing Multi-Dimensional Scaling of Poisson Distance for all samples except for outlier

The raw counts data was normalized again by calculating the fragments per million mapped fragments (FPM), this is equal to CPM (counts per million). The log2 was calculated of these given values. 
```{r}
counts.fpm <- log2( (counts / (colSums(counts) / 1e6)) + 1 )
```

A new group is made in the annotation data frame which contains information about the condition of the samples (control or schizophrenic), which hippocampal subfield is examined and whether the patient was on anti-psychotic medication or not. This is done to make it easier to compare these different groups.

Using DESeq a comparison is made between the schizophrenic patients who were on anti-psychotic medication and those who were not. A comparison is also made between the schizophrenia group off medication and the healthy control group of each subfield. These two comparisons are made for each hippocampal subfield using an adjusted p-value of 0.05. The results of these comparisons give information about the amount of genes that are up or down regulated.

```{r}
annotation$Group <- paste(annotation$Condition, annotation$Region, annotation$Drug, sep="_")
dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = annotation,
                              design = ~ 0 + Group)
dds <- DESeq(dds)
resultsNames(dds)
res_CA1_drug <- results(dds, contrast = c("Group", "schizophrenia_CA1_ON", "schizophrenia_CA1_OFF"), alpha = 0.05)
summary(res_CA1_drug)

res_CA1_SZ_CTL <- results(dds, contrast = c("Group", "schizophrenia_CA1_OFF", "control_CA1_OFF"), alpha = 0.05)
summary(res_CA1_SZ_CTL)

res_CA3_drug <- results(dds, contrast = c("Group", "schizophrenia_CA3_ON", "schizophrenia_CA3_OFF"), alpha = 0.05)
summary(res_CA3_drug)

res_CA3_SZ_CTL <- results(dds, contrast = c("Group", "schizophrenia_CA3_OFF", "control_CA3_OFF"), alpha = 0.05)
summary(res_CA3_SZ_CTL)

res_DG_drug <- results(dds, contrast = c("Group", "schizophrenia_DG_ON", "schizophrenia_DG_OFF"), alpha = 0.05)
summary(res_DG_drug)

res_DG_SZ_CTL <- results(dds, contrast = c("Group", "schizophrenia_DG_OFF", "control_DG_OFF"), alpha = 0.05)
summary(res_DG_SZ_CTL)
```

For each hippocampal subfield, a volcano plot is made using the results from DESeq to show the difference in gene expression between schizophrenic patient who were on medication and those who were not. In the volcano plot all genes are visualized but only the DEG's are highlighted. To determine which gene is a DEG, a threshold of 0.05 is set for the adjusted p-value and 2 for the fold-change (FC).
```{r}
library(EnhancedVolcano)
library(ggrepel)

## Simple function for plotting a Volcano plot, returns a ggplot object
deseq.volcano <- function(res, datasetName) {
  return(EnhancedVolcano(res, x = 'log2FoldChange', y = 'padj',
                         lab=rownames(res),
                         title = paste(datasetName, ""),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         # Change text and icon sizes
                         labSize = 3, pointSize = 1.5, axisLabSize=10, titleLabSize=12,
                         subtitleLabSize=8, captionLabSize=10,
                         # Disable legend
                         legendPosition = "none",
                         # Set cutoffs
                         pCutoff = 0.05, FCcutoff = 2))
}

deseq.volcano(res = res_CA1_drug, datasetName = "CA1")
```
Figure 5: Volcano plot showing the gene expression of subfield CA1 resulting from the comparison of patients on and off drugs
The expression values are plotted with the -log10 p-value on the y-axis and the log fold-change on the x-axis. The plot is separated into according to the set thresholds with a different color for each section. The grey area shows the genes with expressions below both thresholds, the green area shows the genes with expressions only above the FC threshold, the genes with only high p-values are shown in blue, and in red, the significant DEG's or the genes with expressions above both thresholds are shown.

```{r}
deseq.volcano(res = res_CA3_drug, datasetName = "CA3")
```
Figure 6: Volcano plot showing the gene expression of subfield CA3 resulting from the comparison of patients on and off drugs

```{r}
deseq.volcano(res = res_DG_drug, datasetName = "DG")
```
Figure 7: Volcano plot showing the gene expression of subfield DG resulting from the comparison of patients on and off drugs


```{r}
pval_threshold <- 0.05
CA1_drug.degs <- row.names(res_CA1_drug[which(res_CA1_drug$padj <= pval_threshold), ])
CA3_drug.degs <- row.names(res_CA3_drug[which(res_CA3_drug$padj <= pval_threshold), ])
DG_drug.degs <- row.names(res_DG_drug[which(res_DG_drug$padj <= pval_threshold), ])

CA1_SZ_CTL.degs <- row.names(res_CA1_SZ_CTL[which(res_CA1_SZ_CTL$padj <= pval_threshold), ])
CA3_SZ_CTL.degs <- row.names(res_CA3_SZ_CTL[which(res_CA3_SZ_CTL$padj <= pval_threshold), ])
DG_SZ_CTL.degs <- row.names(res_DG_SZ_CTL[which(res_DG_SZ_CTL$padj <= pval_threshold), ])

```


```{r}
library(gplots)
# Create a Venn-diagram given just the list of gene-names for both sets

venn( list(CA1=CA1_drug.degs,CA3=CA3_drug.degs,DG=DG_drug.degs ) )
```
Figure 8: Venn diagram showing the overlap of DEG's between schizophrenia patients ON and OFF medication in the three hippocampal subfields: CA1, CA3 and DG


```{r}
library(VennDiagram)
grid.newpage()
draw.triple.venn(area1 = 100, area2 = 361, area3 = 49, n12 = 4, n23 = 2, n13 = 1, 
    n123 = 0, category = c("DG", "CA3", "CA1"), lty = "blank", scale=T,
    fill = c("skyblue", "pink1", "mediumorchid"))
```

```{r}
venn( list(CA1_SZ_OFF_ON=CA1_drug.degs, CA1_SZ_OFF_CTL=CA1_SZ_CTL.degs ) )
venn( list(CA3_SZ_OFF_ON=CA3_drug.degs, CA3_SZ_OFF_CTL=CA3_SZ_CTL.degs ) )
venn( list(DG_SZ_OFF_ON=DG_drug.degs, DG_SZ_OFF_CTL=DG_SZ_CTL.degs ) )
```
Figure: Venn diagram showing the overlap of DEG's between schizophrenia patients ON and OFF medication and between schizophrenia patients without medication and the healthy control for hippocampal subfield CA1

Figure: Venn diagram showing the overlap of DEG's between schizophrenia patients ON and OFF medication and between schizophrenia patients without medication and the healthy control for hippocampal subfield CA3

Figure: Venn diagram showing the overlap of DEG's between schizophrenia patients ON and OFF medication and between schizophrenia patients without medication and the healthy control for hippocampal subfield DG

sz_CA1_off - ctl_CA1: 57, 2

sz_CA1_on - sz_CA1_off: 1, 48

sz_CA1_on - ctl_CA1: 0, 1


sz_ca3_off - ctl_ca3: 26, 2

sz_ca3_on - sz_ca3_off: 171, 190

sz_ca3_on - ctl_ca3: 31, 358


sz_dg_off - ctl_dg: 11, 3

sz_dg_on - sz_dg_off: 86, 14

sz_dg_on - ctl_dg: 144, 12



go enrichment
```{r}
library(org.Hs.eg.db)
hs <- org.Hs.eg.db
my.symbols <- CA3_drug.degs
ENTREZ <- select(hs, 
       keys = my.symbols,
       columns = c("ENTREZID", "SYMBOL"),
       keytype = "SYMBOL") %>%
  filter(!is.na(ENTREZID))

entrez_ID <- ENTREZ$ENTREZID
```


```{r}
library(DOSE)
edo <- enrichDGN(entrez_ID)
```


```{r}
library(enrichplot)
barplot(edo, showCategory=20) 
```
<<<<<<< HEAD
=======

>>>>>>> d38c595e93fb39de05b4841b24a1b564d0b26ee4

MTRNR2 gene stands out in the volcano plot of each comparison. This is a neuroprotective gene that protects against neuronal death induced by alzheimer. Often shows higher expression in Schizophrenia cases than in control subjects. Was not mentioned in the article.


write.table(CA3_drug.degs, file = "CA3_degs.txt", row.names = F, quote = F)


```{r}
GO_CA1_data <- "CA1_SZ_CTL.txt"
GO_CA1 <- read.delim(GO_CA1_data)
#read.table(file = GO_data, header = TRUE, sep = "\t")
head(GO_CA1)

GO_CA1$log10 <- -log10(GO_CA1$q.value.Bonferroni)

GO_CA1 %>%
  filter(q.value.Bonferroni < 0.05) %>%
  ggplot(aes(x= log10, y= Hit.Count.in.Query.List, color = Category, label = Name))+
  geom_point() +geom_text_repel(hjust=0, vjust=0, size = 3, max.overlaps = Inf, segment.size = 0.2)+
  theme_minimal()
```
```{r}
GO_CA3_data <- "CA3_SZ_CTL.txt"
GO_CA3 <- read.delim(GO_CA3_data)
#read.table(file = GO_data, header = TRUE, sep = "\t")
head(GO_CA3)

GO_CA3$log10 <- -log10(GO_CA3$q.value.Bonferroni)

GO_CA3 %>%
  filter(q.value.Bonferroni < 0.05) %>%
  ggplot(aes(x= log10, y= Hit.Count.in.Query.List, color = Category, label = Name))+
  geom_point() +geom_text_repel(hjust=0, vjust=0, size = 3, max.overlaps = Inf, segment.size = 0.2)+
  theme_minimal()
```

```{r}
GO_DG_data <- "DG_SZ_CTL.txt"
GO_DG <- read.delim(GO_DG_data)
#read.table(file = GO_data, header = TRUE, sep = "\t")
head(GO_DG)

GO_DG$log10 <- -log10(GO_DG$q.value.Bonferroni)

GO_DG %>%
  filter(q.value.Bonferroni < 0.05) %>%
  ggplot(aes(x= log10, y= Hit.Count.in.Query.List, color = Category, label = Name))+
  geom_point() +geom_text_repel(hjust=0, vjust=0, size = 3, max.overlaps = Inf, segment.size = 0.2)+
  theme_minimal()
```


references

Ivanova, E. M., Kandilarova, S. M., Lukanov, T. I., Naumova, E. J., Akabalieva, K. V., & Milanova, V. K. (2021). NGS-based mtDNA Profiling Could Reveal Genetic Alterations in Schizophrenia. Current topics in medicinal chemistry, 21(11), 938–948. https://doi.org/10.2174/1568026621666210521155500